internal/workermanager/discovery_worker.go

```diff
@@ -59,6 +74,69 @@ func (m *Manager) IndexAsset(ctx context.Context, job worker.JobSpec) error {
 	return nil
 }
 
+func (m *Manager) SyncAssets(ctx context.Context, job worker.JobSpec) error {
```

Comment:
@batrov 
Can you help me understand these changes. 

Based on my understanding, for sample service - `bigquery`, this is the flow :

1. Initial state:
index1:  `bigquery` -> `universe` (record count - 10k)

2. Clone from bigquery to bigquery-bak
index1:  `bigquery` -> `universe` (record count - 10k)
index2:  `bigquery-bak` (record count - 10k)

3. Update alias for backup index
index1:  `bigquery` -> `universe` (record count - 10k)
index2:  `bigquery-bak` -> `universe` (record count - 10k)

------------ at this point, when somebody makes a search api call, it will return duplicate results.

4. Delete index - `bigquery`
index2: `bigquery-bak` -> `universe` (record count - 10k)



5. Get all assets from postgres and store it in-memory
bigquery asset count in postgres - 20k
---- the asset count might be huge and lead to OOM memory issues on the pod.

6. Insert records into ES one by one. This will also recreate the `bigquery` index
index1: `bigquery` -> `universe` (record count - 20k)
index2: `bigquery-bak` -> `universe` (record count - 10k)

--- at this point during the asset insertion into ES, the search api will return duplicate results.

7. Delete backup index 
index1: `bigquery` -> `universe` (record count - 20k)

8. Update index settings for `bigquery` index to allow writes
   `{"settings":{"index.blocks.write":false}}`
--- where are we setting this to `true` to block writes initially?

Reply:
> Get all assets from postgres and store it in-memory
bigquery asset count in postgres - 20k
---- the asset count might be huge and lead to OOM memory issues on the pod.

agree, this can be improved further. Was thinking adding a batching logic will help here

> Update index settings for bigquery index to allow writes
{"settings":{"index.blocks.write":false}}
--- where are we setting this to true to block writes initially?

Updated in latest commit, it will be set to true before calling the clone API

Reply:
Lets also see if we can reduce the time during which duplicate results are returned from the search API when this job is running.

Reply:
updated so upsert process should not disturb existing search result